ui/components/src/DateTimeRangePicker/AbsoluteTimePicker.tsx

```diff
@@ -26,32 +26,74 @@ interface AbsoluteTimeFormProps {
   onChange: (timeRange: AbsoluteTimeRange) => void;
 }
 
+type AbsoluteTimeRangeInputValue = {
+  [Property in keyof AbsoluteTimeRange]: string;
+};
+
 export const AbsoluteTimePicker = ({ initialTimeRange, onChange }: AbsoluteTimeFormProps) => {
+  const { formatWithUserTimeZone } = useTimeZone();
+
+  // Time range values as dates that can be used as a time range.
   const [timeRange, setTimeRange] = useState<AbsoluteTimeRange>(initialTimeRange);
+
+  // Time range values as strings used to populate the text inputs. May not
```

Comment:
The core of the bug I'm fixing is that we were trying to translate the text input to a `Date` `onChange`, which would make it impossible to edit the field because it would undo things like removing a character. Fixing this by tracking "dates as strings" and "dates as dates" as separate bits of state.